added the two books that are auto-generated (#45432)
openshift/openshift-docs: contributing_to_docs/doc_guidelines.adoc

In the Atom editor, you can use `Ctrl`+`J` to undo hard wrapping on a paragraph.
====

> * The following books are auto-generated and should not be manually edited:

Reviewer: I'm not sure if this is the right place for this guidance, since this is general guidelines on how to edit files.

I'd consider adding it as it's own new section on auto-generated content. wdyt?

Author: Yes agreed good idea

> ** `cli_reference`

Reviewer: I'd suggest listing them with the book/section name and then the relevant files, because they're different between the two and might not be super obvious. Maybe something like this:

```
* The following content is auto-generated each release and should not be manually edited:

** The OpenShift CLI (`oc`) command references (`modules/oc-by-example-content.adoc` and `modules/oc-adm-by-example-content.adoc`)
** The API references (anything under the `rest_api` folder that is in a `<topic>_apis` subfolder)
```

@jboxman-rh Can you confirm that this is correct? Technically there are some files under rest_api/ that can be edited I believe, so I didn't want to make a blanket statement that the entire folder was off limits.

Reviewer 2: More or less. These as well:

- https://docs.openshift.com/container-platform/4.10/rest_api/objects/index.html
- https://docs.openshift.com/container-platform/4.10/rest_api/index.html

Thanks.

** `rest-api`

> 

Reviewer: As @abhatt-rh recommended, we could put something afterward here to help people with what to do if something needs updated in either of these files. Maybe something like:

```
If content in these files need to be updated, the update must be made in the applicable code repository where these docs are generated from. The updates will then be reflected the next time the docs are generated; likely a future release. For help with where to make the updates, contact Andrea Hoffer for the CLI docs or Jason Boxman for the API docs.
```
Could maybe even add a link to me and @jboxman-rh's GitHub accounts in there?

For more information about creating text snippets, see the
link:https://redhat-documentation.github.io/modular-docs/#using-text-snippets[Red Hat modular docs reference guide]

[id="Auto-generated-content"]
== Auto-generated content

> * The following content is auto-generated in each release and should not be manually edited:

Reviewer 3: ```suggestion
* The following content is auto-generated in each release and must not be manually edited:
```
I also suggest make this a statement, not a list item, and changing the others to first-level list items.

> ** The Openshift CLI (`oc`) command references `modules/oc-by-example-content.adoc` and `modules/oc-adm-by-example-content.adoc`.

Reviewer: Openshift -> OpenShift

> ** The {product-title} CLI (`oc`) command references `modules/oc-by-example-content.adoc` and `modules/oc-adm-by-example-content.adoc`.

Reviewer: ```suggestion
** The OpenShift CLI (`oc`) command references `modules/oc-by-example-content.adoc` and `modules/oc-adm-by-example-content.adoc`.
```

Just a reminder not to use the "Commit suggestion" button for this :)

Author: ok noted :)

> ** The API references content in the `rest_api` folder that comprises of `<topic>_apis` subfolders. It also includes the `rest_api/objects/index.adoc` and `rest_api/index.adoc` assemblies.

Reviewer 3: Maybe:

```suggestion
** The following API references content in the `rest_api` folder: the contents of all `<topic>_apis` subfolders and the `rest_api/objects/index.adoc` and `rest_api/index.adoc` assemblies.
```

[NOTE]
====

> If the content in these files needs to be updated, the update must be made in the applicable code repository where these files are generated from. The updates are reflected when the files are generated the next time, for example a future release. For help with where to make the updates, you can contact https://github.com/bergerhoffer[Andrea Hoffer] for the CLI docs or https://github.com/jboxman[Jason Boxman] for the API docs.

Reviewer 2: ```suggestion
If the content in these files needs to be updated, the update must be made in the applicable code repository where these files are generated from. The updates are reflected when the files are generated the next time, for example a future release. For help with where to make the updates, you can contact https://github.com/bergerhoffer[Andrea Hoffer] for the CLI docs or https://github.com/jboxman-rh[Jason Boxman] for the API docs.
```
